apps/web/src/components/NextRoleAndCareerObjective/CareerObjectiveDialog.tsx

```diff
@@ -0,0 +1,81 @@
+import { ReactNode, useState } from "react";
+import { useIntl } from "react-intl";
+
+import { FragmentType, getFragment, graphql } from "@gc-digital-talent/graphql";
+import { commonMessages } from "@gc-digital-talent/i18n";
+import { Button, Dialog } from "@gc-digital-talent/ui";
+
+import CareerObjective from "./CareerObjective";
+
+export const CareerObjectiveDialog_Fragment = graphql(/* GraphQL */ `
+  fragment CareerObjectiveDialog on User {
+    firstName
+    employeeProfile {
+      ...CareerObjectiveInfo
+    }
+  }
+`);
+
+interface CareerObjectiveDialogProps {
+  careerObjectiveDialogQuery: FragmentType<
+    typeof CareerObjectiveDialog_Fragment
+  >;
+  trigger?: ReactNode;
+  defaultOpen?: boolean;
+}
+
+const CareerObjectiveDialog = ({
+  careerObjectiveDialogQuery,
+  trigger,
+  defaultOpen = false,
+}: CareerObjectiveDialogProps) => {
+  const intl = useIntl();
+  const [isOpen, setOpen] = useState<boolean>(defaultOpen);
+  const careerObjective = getFragment(
+    CareerObjectiveDialog_Fragment,
+    careerObjectiveDialogQuery,
+  );
+
+  const title =
+    `${careerObjective.firstName}'s ` +
+    intl.formatMessage({
+      defaultMessage: "career objective",
```

Comment:
Might want to have a fallback not provided as firstName is nullable

Reply:
Fixed in [Fix career objective dialog preview list title](https://github.com/GCTC-NTGC/gc-digital-talent/pull/13300/commits/9c730a5dce76da4087ec5103e4ae8aa2bc378cef)